Firmware/ultralcd.cpp

```diff
@@ -3712,6 +3712,16 @@ void lcd_mesh_calibration_z()
   lcd_return_to_status();
 }
 
+#ifdef TEMP_MODEL
+void lcd_temp_model_cal()
+{
+  enquecommand_P(PSTR("M310 A F0"));
+  enquecommand_P(PSTR("M310 S1"));
```

Comment:
In case calibration fails, M310 S1 will refuse to enable the model (which is fine), however M500 will still trigger an eeprom save. If the previous calibration succeeded, this will result in the model being permanently disabled.

Reply:
Okay then I need another approach with lcd_command_step or so.